src/Mustache/Compiler.php

```diff
@@ -322,7 +328,11 @@ private function block($nodes)
     }
 
     const SECTION_CALL = '
-        $value = $context->%s(%s);%s
+        try {
+            $value = $context->%s(%s);%s
+        } catch (Mustache_Exception_UnknownVariableException $ex) {
+            $value = "";
```

Comment:
why aren't sections (and inverted sections) strict?

Reply:
The PR is a bit old so I have forgotten a bit about it. I will try to deep dive in again next week but this is likely due to some previous comments:
https://github.com/bobthecow/mustache.php/pull/325#discussion_r141260585
https://github.com/bobthecow/mustache.php/pull/325#issuecomment-331433290

Reply:
gotcha. i'll poke around a bit and see if it makes sense to me.